modules/external-api/src/main/java/org/opencastproject/external/util/CaptureAgentUtils.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+package org.opencastproject.external.util;
+
+import static com.entwinemedia.fn.data.json.Jsons.arr;
+import static com.entwinemedia.fn.data.json.Jsons.f;
+import static com.entwinemedia.fn.data.json.Jsons.obj;
+import static com.entwinemedia.fn.data.json.Jsons.v;
+import static org.opencastproject.util.DateTimeSupport.toUTC;
+
+import org.opencastproject.capture.CaptureParameters;
+import org.opencastproject.capture.admin.api.Agent;
+
+import com.entwinemedia.fn.data.json.Field;
+import com.entwinemedia.fn.data.json.JValue;
+import com.entwinemedia.fn.data.json.Jsons;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class CaptureAgentUtils {
+
+  private static final String JSON_KEY_AGENT_ID = "agent_id";
+  private static final String JSON_KEY_STATUS = "status";
+  private static final String JSON_KEY_URL = "url";
+  private static final String JSON_KEY_INPUTS = "inputs";
+  private static final String JSON_KEY_UPDATE = "update";
+
+
+  private CaptureAgentUtils() {
+  }
+
+  /**
+   * Generate a JSON Object for the given capture agent
+   *
+   * @param agent
+   *          The capture agent
+   * @return A {@link JValue} representing the capture agent
+   */
+  public static JValue generateJsonAgent(Agent agent) {
+    List<Field> fields = new ArrayList<>();
+    String devices = (String) agent.getCapabilities().get(CaptureParameters.CAPTURE_DEVICE_NAMES);
+    fields.add(f(JSON_KEY_STATUS, v(agent.getState(), Jsons.BLANK)));
+    fields.add(f(JSON_KEY_AGENT_ID, v(agent.getName())));
+    fields.add(f(JSON_KEY_UPDATE, v(toUTC(agent.getLastHeardFrom()), Jsons.BLANK)));
+    fields.add(f(JSON_KEY_URL, v(agent.getUrl(), Jsons.BLANK)));
+    fields.add(f(JSON_KEY_INPUTS, (StringUtils.isEmpty(devices)) ? arr() : arr(devices.split(","))));
```

Comment:
The timezones reported by capture agents should not be relevant to API clients as the backend is supposed to ensure the correct handling of timezones for scheduling.

Reply:
It might be if they're building a scheduling UI over top and need to support multiple timezones.  For instance, if I'm in UTC-6, and scheduling an agent in UTC+2, it might be helpful to *know* that.

I suppose this comes down to the bugs in the current scheduling UI re: timezones.  Perhaps we leave this alone and if we find we need it we can add it in API 1.2.0.

Reply:
In case somebody wants to add (parts of) the capture agent configuration, it could be added to 1.1.0 (before r/6.x is cut) or in a future API version. 
It would require some thoughts (if done properly) as it is not that clear (at least to me) which parts of the capture agent configuration are "standard", "legacy" or "vendor" - and just adding all would unnecessarily blow up this response which again is undesirable.

Anyway, this PR makes that easier as it is now quite obvious how versioning works as the code supports multiple versions.

As for us, though, we simply don't have that requirement.